functions/that-api-webhooks/hooks/send/activeCampaign.js

```diff
@@ -0,0 +1,245 @@
+
+const dlog = debug('that:api:webhooks:activeCampaign');
+
+// Create / Update contact:
+// 200 - exists and updated
+// 201 - created
+// result set is the same
+
+let acBaseUrl = envConfig.activeCampaignApi;
+if (acBaseUrl.endsWith('/'))
+  acBaseUrl = acBaseUrl.substring(0, acBaseUrl.length - 1);
+const axiosReqConfig = {
+  headers: {
+    'Api-Token': envConfig.activeCampaignKey,
+  },
+};
+
+function findContactByEmail(email) {
+  dlog('call findContactByEmail for %s', email);
+  const url = `${acBaseUrl}/contacts`;
+  const reqConfig = {
+    url,
+    method: 'get',
+    ...axiosReqConfig,
+    params: {
+      email,
+    },
+  };
+
+  return axios(reqConfig)
+    .then(result => {
+      if (result.status !== 200) {
+        dlog('Non 200 result from query email %s', email);
+        Sentry.withScope(scope => {
+          scope.setLevel('warning');
+          scope.setContext(
+            'non 200 status from contact query',
+            { email },
+            result,
+          );
+          Sentry.captureMessage('non 200 status from contact query');
+        });
+        return undefined;
+      }
+      if (
+        result.data &&
+        result.data.contacts &&
+        result.data.contacts.length > 1
+      ) {
+        dlog(
+          'query for email, %s, returned %d matches, expect 1 or 0',
+          email,
+          result.data.contacts.length,
+        );
+        Sentry.withScope(scope => {
+          scope.setLevel('error');
+          scope.setContext(
+            'query by email return > 1 contact',
+            { email },
+            result.data.contacts,
+          );
+          Sentry.captureMessage('query by email returned > 1 contact');
+        });
+        return undefined;
+      }
+      dlog(`returning ${result.data.contacts[0]}`);
+      return result.data.contacts[0];
+    })
+    .catch(err => {
+      dlog('exception looking for contact: %s', err);
+      Sentry.captureException(err);
+      return undefined;
+    });
+}
+
+function createContact(contact) {
+  // https://developers.activecampaign.com/reference#create-a-contact-new
+  /* minimum required for payload
+    {
+      contact: {
+        email: email@email.com,
+      }
+    }
+  */
+  dlog('call createContact for %o', contact);
+  const url = `${acBaseUrl}/contacts`;
+  const reqConfig = {
+    url,
+    method: 'post',
+    ...axiosReqConfig,
+    data: contact,
+  };
+  return axios(reqConfig)
+    .then(r => {
+      if (![200, 201].includes(r.status)) {
+        dlog('exception creating contact');
+        Sentry.withScope(scope => {
+          scope.setLevel('error');
+          scope.setContext('exception creating contact', contact, r);
+          Sentry.captureMessage('exception creating contact');
+        });
+        return undefined;
+      }
+
+      return r.data.contact;
+    })
+    .catch(err => {
+      dlog('exception creating contact: %s', err);
+      Sentry.captureException(err);
+      return undefined;
```

Comment:
I'm seeing this a few places.. Why not just log and rethrow? Returning undefined smells a bit for me here..

Reply:
corrected by adding better error handling